shared/teams/channel/tabs/index.tsx

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react'
+import * as Types from '../../../constants/types/teams'
+import * as ChatTypes from '../../../constants/types/chat2'
+import * as Kb from '../../../common-adapters'
+import flags from '../../../util/feature-flags'
+import {
+  globalColors,
+  globalMargins,
+  globalStyles,
+  isMobile,
+  platformStyles,
+  styleSheetCreate,
+} from '../../../styles'
+
+export type TabKey = 'members' | 'attachments' | 'bots' | 'settings' | 'loading'
+
+export type OwnProps = {
+  teamID: Types.TeamID
+  conversationIDKey: ChatTypes.ConversationIDKey
+  selectedTab: TabKey
+  setSelectedTab: (t: TabKey) => void
+}
+
+export type Props = OwnProps & {
+  admin: boolean
+  error?: string
+  loadBots: () => void
+  loading: boolean
+}
+
+const TabText = ({selected, text}: {selected: boolean; text: string}) => (
+  <Kb.Text type="BodySmallSemibold" style={selected ? styles.tabTextSelected : styles.tabText}>
+    {text}
+  </Kb.Text>
+)
+
+const ChannelTabs = (props: Props) => {
+  const {selectedTab, setSelectedTab} = props
+  const wrapTab = (key: string, child: React.ReactNode) => (
+    <Kb.Box key={key} style={styles.tabTextContainer}>
+      {child}
+    </Kb.Box>
+  )
+  const tabs = [
+    wrapTab('members', <TabText selected={selectedTab === 'members'} text="Members" />),
+    wrapTab('attachments', <TabText selected={selectedTab === 'attachments'} text="Attachments" />),
+    ...(flags.botUI ? [wrapTab('bots', <TabText selected={selectedTab === 'bots'} text="Bots" />)] : []),
+    ...(props.admin
+      ? [wrapTab('settings', <TabText selected={selectedTab === 'settings'} text="Settings" />)]
+      : []),
+    ...(!isMobile && props.loading
+      ? [<Kb.ProgressIndicator key="loading" style={styles.progressIndicator} />]
+      : []),
+  ]
+
+  const onSelect = (tab: any) => {
```

Comment:
Should this `any` be `TabKey`?

Reply:
it's actually a ReactNode, and if I try to type it, it gets really messy. I cleaned it up a bit but this `onSelect` has to stay `any`.

Reply:
I made a ticket to improve typing on `Tabs`

Reply:
What's the proposal?

Reply:
OK I made a ticket to refactor Kb.Tabs.